Hardcore.lua

```diff
@@ -340,6 +348,26 @@ function Hardcore:PLAYER_ENTERING_WORLD()
 	if( not C_ChatInfo.IsAddonMessagePrefixRegistered(COMM_NAME) ) then
 		C_ChatInfo.RegisterAddonMessagePrefix(COMM_NAME)
 	end
+
+  -- Hook TargetFrame classification and warn if PvP enabled and enemy faction 
+  hooksecurefunc("TargetFrame_CheckClassification",function(self, lock)
+    if Hardcore_Settings.grief_warning_conditions == PVP then
+      if UnitIsPVP("target") and UnitGUID("target") ~= PLAYER_GUID then
+        local target_name, _ = UnitName("target")
+        Hardcore:ShowAlertFrame("hc_red", "Target " .. target_name .. " is PvP enabled!")
+      end
+    elseif Hardcore_Settings.grief_warning_conditions == PVP_AND_ENEMY_FACTION then
+      if UnitGUID("target") ~= PLAYER_GUID and UnitIsPVP("target")  then 
+        local faction, _ = UnitFactionGroup("target")
+        if faction ~= nil then
+          if (faction == "Horde" and PLAYER_FACTION == "Alliance") or (faction == "Alliance" and PLAYER_FACTION == "Horde") then
```

Comment:
nit: Wouldn't a not equal condition suffice here?

```
local target_faction, _ = UnitFactionGroup("target")
if target_faction ~= PLAYER_FACTION then
 // do stuff
end
```

Reply:
I think that you are right; I'll test it out. I'm not sure why I originally did this. I might have been worried that target faction could be something other than horde, alliance, or nil